rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -107,6 +113,11 @@ public J visit(@Nullable Tree tree, Integer integer) {
                 .visit(changing, 0, parentCursor);
     }
 
+    @Incubating(since = "7.38.0")
+    public boolean matches(Cursor cursor, JavaCoordinates coordinates) {
```

Comment:
Would suggest dropping this coordinates argument and always using the "replace" coordinate (weird name for matching, but at least hides the weirdness for now)

Reply:
If you ever want to match on a subelement of the thing pointed to by `getCursor()` we could provide separate utility on `Cursor` like:

```java
Cursor c = getCursor();
c.findCursor(binary.getLeft());
```